x/accesscontrol/keeper/keeper.go

```diff
@@ -156,21 +156,95 @@ func (k Keeper) GetWasmDependencyAccessOps(ctx sdk.Context, contractAddress sdk.
 		return nil, err
 	}
 
-	// TODO: extend base access op with message-specific ops once message type identifier is passed in
-	selectedAccessOps, err := k.BuildSelectorOps(ctx, contractAddress, dependencyMapping.BaseAccessOps, senderBech, msgBody, circularDepLookup)
+	accessOps := dependencyMapping.BaseAccessOps
+	specificAccessOpsMapping := []*acltypes.WasmAccessOperations{}
+	if msgInfo.MessageType == acltypes.WasmMessageSubtype_EXECUTE && dependencyMapping.ExecuteAccessOps != nil {
+		specificAccessOpsMapping = dependencyMapping.ExecuteAccessOps
+	} else if msgInfo.MessageType == acltypes.WasmMessageSubtype_QUERY && dependencyMapping.QueryAccessOps != nil {
+		specificAccessOpsMapping = dependencyMapping.QueryAccessOps
+	}
+	for _, specificAccessOps := range specificAccessOpsMapping {
+		if specificAccessOps.MessageName == msgInfo.MessageName {
+			accessOps = append(accessOps, specificAccessOps.WasmOperations...)
+		}
+	}
+
+	selectedAccessOps, err := k.BuildSelectorOps(ctx, contractAddress, accessOps, senderBech, msgInfo, circularDepLookup)
 	if err != nil {
+		// TODO: should we reset wasm dependency mapping here?
```

Comment:
cc @codchen do you think we should be resetting the wasm dependencies if we get an error as part of building selector ops or contract references?

Reply:
the error could be from a bad message body rather than a bad dependency registration, in which case we probably don't want to reset